ring/README.md

```diff
@@ -0,0 +1,14 @@
+# ring
+
+Striped, lossy ring buffers meant for MPSC work.
+
+## benchmarks
+
+Benchmarks are ran on an Intel i7-8700K (3.7GHz 6-core).
+
+| Stripes | Size | Parallel | Ops/Second  | Allocs/Op |
+|:-------:|:----:|:--------:|:------------|:----------|
+| 1       | 128  |          | 131,140,000 | 0 |
+| 16      | 128  |          | 97,960,000  | 0 |
+| 1       | 128  | ✔        | 32,440,000  | 0 |
+| 16      | 128  | ✔        | 47,720,000  | 0 |
```

Comment:
It is concerning that this goes down as the number of stripes increases. Ideally it would scale linearly up to the number of cpus. Ideal N cpus => N stripes, then some extra due to contention (thread migration, hash collisions). I suspect that your striping algorithm could be refined.

Reply:
The parallel benchmarks are ran using an undefined number of goroutines. The Go team refuses to expose thread information, [despite using it in the standard library](https://github.com/golang/go/issues/18590#issuecomment-271661760). I'd much prefer to use the thread id for picking stripes, so right now I'm just randomly picking until I can find a more efficient way.

Reply:
There is this cgo hack:
https://github.com/davecheney/junk/tree/master/id

In other systems one often has a fallback chain, testing if a low-level hack is supported and degrading until a language approved version works. Then you can fix when optimizations stop working without breaking applications.